python/__init__.py

```diff
@@ -72,6 +72,27 @@ def __hash__(self):
 except ImportError:
     pass
 
+# Define a `with edit(layer)` statement
+
+class edit:
+    def __init__(self,layer):
+        self.layer = layer
+
+    def __enter__(self):
+        assert self.layer.startEditing()
+        return self.layer
+
+    def __exit__(self, ex_type, ex_value, traceback):
+        if ex_type is None:
+            assert self.layer.commitChanges()
```

Comment:
You could but you don't really gain anything I don't think unless we raise
it from inside the sip bindings, might as well define it in Python to make
it easier.

On Wed, Aug 12, 2015 at 9:55 PM, Matthias Kuhn <notifications@github.com>
wrote:

> In python/__init__.py
> <https://github.com/qgis/QGIS/pull/2242#discussion_r36850986>:
>
> > @@ -72,6 +72,27 @@ def __hash__(self):
> >  except ImportError:
> >      pass
> >
> > +# Define a `with edit(layer)` statement
> > +
> > +class edit:
> > +    def __init__(self,layer):
> > +        self.layer = layer
> > +
> > +    def __enter__(self):
> > +        assert self.layer.startEditing()
> > +        return self.layer
> > +
> > +    def __exit__(self, ex_type, ex_value, traceback):
> > +        if ex_type is None:
> > +            assert self.layer.commitChanges()
>
> Wouldn't it be nice to have that defined directly in qgsvectorlayer.sip?
> This way we get it for free here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qgis/QGIS/pull/2242/files#r36850986>.
>